### DISCUSSION TITLE
How to bind in DataGridTextColumn.HeaderTemplate

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#14868] [Q&A]
I have a DataGrid with tooltips on column headers:

```
        <DataGridTextColumn Binding="{Binding cellValue}">
          <DataGridTextColumn.HeaderTemplate>
            <DataTemplate>
                <Label Content="header text" ToolTip.Tip="tooltip text"/>
            </DataTemplate>
          </DataGridTextColumn.HeaderTemplate>
        </DataGridTextColumn>
```

How can I bind a value for "header text" to my view model?

I tried 

                <Label Content="{Binding HeaderText}" ToolTip.Tip="tooltip text"/>

but it comes up empty.
Binding the DataGridTextColumn header without the header template works just fine (but of course there is no tooltip then):

        <DataGridTextColumn Header="{Binding HeaderText}" Binding="{Binding cellValue}">

How do I bind the label content in the headertemplate?

**Comment by timunie** [comment]
```xml
<DataGridTextColumn Header="{Binding HeaderText}" Binding="{Binding cellValue}">
 <DataGridTextColumn.HeaderTemplate>
    <DataTemplate>
        <Label Content="{Binding}" ToolTip.Tip="tooltip text"/>
    </DataTemplate>
  </DataGridTextColumn.HeaderTemplate>
</DataGridTextColumn>
```

**Reply by wgraaf** [reply]
That works, thanks,

Could you help me understand why this works?
Apparently the data context for the label is not the data context for the DataGridTextColumn.
How does it bind to the column header text? Is that the default?
Could it bind to other properties of the data context?

**Reply by timunie** [reply]
It's the MainViewModel for the header, not any ItemViewModel
